src/csharp/Microsoft.Spark/Interop/Ipc/JsonSerDe.cs

```diff
@@ -2,70 +2,155 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Linq;
-using Newtonsoft.Json.Linq;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
 
 namespace Microsoft.Spark.Interop.Ipc
 {
     /// <summary>
-    /// Json.NET Serialization/Deserialization helper class.
+    /// System.Text.Json Serialization/Deserialization helper class.
     /// </summary>
     internal static class JsonSerDe
     {
+
         /// Note: Scala side uses JSortedObject when parsing JSON, so the properties
-        /// in JObject need to be sorted.
+        /// in JsonElement need to be sorted.
+        /// <summary>
+        /// Extension method to sort items in a JSON object by keys. Please note that this 
+        /// method uses recursion and shouldn't be used with untrusted/unbounded input data.
```

Comment:
nit: Instead of putting this in the `summary` tag, move it to a `remark` tag, leaving the `summary` tag as is.

```C#
/// <remarks>
/// Please note that this method uses recursion and shouldn't be used with 
/// non-trusted/unbounded input data.
/// </remarks>
```

Reply:
Ah yes, forgot about `<remarks>`, cheers